src/Draco.Compiler/Internal/Symbols/Metadata/MetadataMethodSymbol.cs

```diff
@@ -109,4 +127,73 @@ private void BuildSignature()
+        foreach (var impl in type.GetMethodImplementations())
+        {
+            var implementation = this.MetadataReader.GetMethodImplementation(impl);
+            var body = implementation.MethodBody.Kind switch
+            {
+                HandleKind.MethodDefinition => this.GetFunctionFromDefinition((MethodDefinitionHandle)implementation.MethodBody),
+                HandleKind.MemberReference => this.GetFunctionFromReference((MemberReferenceHandle)implementation.MethodBody),
+                _ => throw new InvalidOperationException(),
+            };
+
+            if (body is null) return null;
+
+            if (!implementation.MethodDeclaration.IsNil && body.CanBeOverriddenBy(this)) 
+            {
+                return implementation.MethodDeclaration.Kind switch
+                {
+                    HandleKind.MethodDefinition => this.GetFunctionFromDefinition((MethodDefinitionHandle)implementation.MethodDeclaration),
+                    HandleKind.MemberReference => this.GetFunctionFromReference((MemberReferenceHandle)implementation.MethodDeclaration),
+                    _ => throw new InvalidOperationException(),
+                };
+            }
+        }
+        return null;
+    }
+
+    private FunctionSymbol? GetFunctionFromDefinition(MethodDefinitionHandle methodDef)
+    {
+        var definition = this.MetadataReader.GetMethodDefinition(methodDef);
+        var provider = new TypeProvider(this.Assembly.Compilation);
+        var signature = definition.DecodeSignature(provider, this);
+        var type = provider.GetTypeFromDefinition(this.MetadataReader, definition.GetDeclaringType(), 0);
+        var symbols = type.DefinedMembers.Concat((type as IMetadataClass)!.PropertyAccessors).OfType<FunctionSymbol>();
+        foreach (var function in symbols)
+        {
+            if (function.Name != this.MetadataReader.GetString(definition.Name)) continue;
+            if (CompareSignatures(function, signature)) return function;
+        }
+        return null;
+    }
+
+    private FunctionSymbol? GetFunctionFromReference(MemberReferenceHandle methodRef)
+    {
+        var reference = this.MetadataReader.GetMemberReference(methodRef);
+        var provider = new TypeProvider(this.Assembly.Compilation);
+        var signature = reference.DecodeMethodSignature(provider, this);
+        var type = provider.GetTypeFromReference(this.MetadataReader, (TypeReferenceHandle)reference.Parent, 0);
+        var symbols = type.DefinedMembers.Concat((type as IMetadataClass)!.PropertyAccessors).OfType<FunctionSymbol>();
+        foreach (var function in symbols)
+        {
+            if (function.Name != this.MetadataReader.GetString(reference.Name)) continue;
+            if (CompareSignatures(function, signature)) return function;
+        }
+        return null;
+    }
+
+    private static bool CompareSignatures(FunctionSymbol function, MethodSignature<TypeSymbol> signature)
```

Comment:
Suggest a name to reflect what this returns (like `SignaturesMatch` or `SignaturesAreEqual`)